torch/csrc/autograd/engine.cpp

```diff
@@ -340,7 +340,7 @@ static void validate_outputs(const edge_list& edges, variable_list& grads, const
         ss << metadata.shape();
         AT_ERROR(format_error(ss.str()));
       }
-      grads[i] = at::sum_to(grads[i], metadata.shape());
+      grads[i] = at::sum_to(std::move(grads[i]), metadata.shape());
```

Comment:
Is `grads[i]` a value or a reference? `grads` (the input to this function) gets reused after the invocation to this function (line 381), so does this invalidate the entry?

Reply:
`grads[i]` is a reference, because otherwise it would already be an rvalue, and the `std::move` would be unnecessary. It invalidates the entry for the time of calling into `sum_to`, and replaces it immediately after, so I think it preserves the current behavior.